File: data/mi1/rooms/mancomb.lua
roomDefinition = {
	player_pos = {300,0,0}
}

room = scumm.factory.dialogue_room (roomDefinition)

--room:add_asset(sprites.mancomb2)

room:add( {
	{ pos = {0, 0, -3}, components = { { type="gfx", image="gfx/mancomb.png" }}},
	--factory.objc { id = "mancomb.mancomb" },
})

-- local d = strings.dialogues.mancomb

-- function room.afterstartup() 
--  	local l = variables.first_time_mancomb and {d[1], d[2]} or {d[8]}
--  	variables.first_time_mancomb = false
-- 	local actions = {
--  		{ type = action.say, args = {actor="mancomb.mancomb", lines = l, animate = false}},
-- 		{ type = action.start_dialogue, args ={ dialogue="mancomb"}}
-- 	}
-- 	local s = ms2(actions)
-- 	monkey.play(s)
-- end

File: data/mi1_py/rooms/mancomb.py
from lib_py.scumm.room import RoomDialogue
from lib_py.script import Script
import lib_py.scumm.entity as se
import lib_py.scumm.actions as sa
import lib_py.engine as engine
import example

st = engine.data['strings']
sl = st['dialogues']['mancomb']

def f():
    s = Script()
    s.addAction (sa.Say(lines = [sl[1], sl[2]], tag ='mancomb.mancomb'))
    s.addAction (sa.StartDialogue (dialogueId='mancomb'))
    example.play(s)


def builder():
    r = RoomDialogue(id='scummbar')
    r.add (se.BackgroundItem(image='gfx/mancomb.png', pos = [0,0,-1]), 'main')
    r.addItem (id = 'mancomb.mancomb')
    #r.addItem (id = 'guybrush.dialogue')
    r.init.append(f)
    return r

engine.addRoom (id = 'mancomb', f=builder)
